mlflow/sklearn/__init__.py

```diff
@@ -1536,40 +1616,45 @@ def out(*args, **kwargs):
 
     _apply_sklearn_descriptor_unbound_method_call_fix()
 
-    for class_def in _gen_estimators_to_patch():
+    if flavor_name == "xgboost_sklearn":
```

Comment:
Following from https://github.com/mlflow/mlflow/pull/5078/files#r752577834, can we refer to `mlflow.xgboost.FLAVOR_NAME` instead of `xgboost_sklearn`?